src/simulation/grids.jl

```diff
@@ -0,0 +1,352 @@
+- `mesh_warp_type`: Mesh warping type ([`SLEVEWarp`](@ref) or [`LinearWarp`](@ref))
+- `topo_smoothing`: Apply topography smoothing
+"""
+function PlaneGrid(
+    FT;
+    context = ClimaComms.context(),
+    x_elem = 6,
+    x_max = 300000.0,
+    z_elem = 10,
+    z_max = 30000.0,
+    nh_poly = 3,
+    z_stretch = true,
+    dz_bottom = 500.0,
+    bubble = false,
+    periodic_x = true,
+    topography::AbstractTopography = NoTopography(),
+    topography_damping_factor = 5.0,
+    mesh_warp_type::MeshWarpType = LinearWarp(),
+    topo_smoothing = false,
+)
+    n_quad_points = nh_poly + 1
+
+    stretch =
+        z_stretch ? Meshes.HyperbolicTangentStretching{FT}(dz_bottom) : Meshes.Uniform()
+
+    hypsography_fun = hypsography_function_from_topography(
+        FT, topography, topography_damping_factor, mesh_warp_type, topo_smoothing,
+    )
+
+    z_mesh = DefaultZMesh(
+        FT;
+        z_min = 0,
+        z_max,
+        z_elem,
+        stretch,
+    )
+
+    grid = SliceXZGrid(
+        FT;
+        z_elem, x_elem, x_min = 0, x_max, z_min = 0, z_max, z_mesh,
+        periodic_x,
+        n_quad_points,
+        device = ClimaComms.device(context),
+        context,
+        stretch,
+        hypsography_fun,
+        global_geometry = Geometry.CartesianGlobalGeometry(),
+    )
+
+    return grid
+end
+
+"""
+    hypsography_function_from_topography(
+        FT, topography, topography_damping_factor, mesh_warp_type, topo_smoothing)
+
+Create a hypsography function that handles topography integration.
+"""
+function hypsography_function_from_topography(
+    FT::Type{<:AbstractFloat},
+    topography::AbstractTopography,
+    topography_damping_factor,
+    mesh_warp_type::MeshWarpType,
+    topo_smoothing,
+)
+    return function (h_grid, z_grid)
+        topography isa NoTopography && return Hypsography.Flat()
+
+        # Create horizontal space to work with topography
+        h_space = if h_grid isa Grids.SpectralElementGrid1D
+            Spaces.SpectralElementSpace1D(h_grid)
+        elseif h_grid isa Grids.SpectralElementGrid2D
+            Spaces.SpectralElementSpace2D(h_grid)
+        else
+            error("Unsupported horizontal grid type $(typeof(h_grid))")
+        end
+
+        # Load topography data
+        if topography isa EarthTopography
+            context = ClimaComms.context(h_space)
+            z_surface = SpaceVaryingInput(
+                AA.earth_orography_file_path(; context),
+                "z",
+                h_space,
+            )
+            @info "Remapping Earth orography from ETOPO2022 data onto horizontal space"
+        else
+            z_surface = SpaceVaryingInput(topography_function(topography), h_space)
+            @info "Using $(nameof(typeof(topography))) orography"
+        end
+
+        # Apply topography-specific processing
+        if topography isa EarthTopography
+            mask(x::FT) where {FT} = x * FT(x > 0)
+            z_surface = @. mask(z_surface)
```

Comment:
```suggestion
            @. z_surface = min(z_surface, 0)
```
Writing this in terms of `min` will probably reduce latency, and there's no need to allocate a new `z_surface` field.

But more importantly, if the actual topography data includes negative values (I'm guessing landlocked areas below sea level), shouldn't we account for them in the final diffused surface elevation? It feels strange to just clip them to zero.